src/legacy/core_plugins/kibana/public/context/api/context.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// @ts-ignore
+import { SearchSourceProvider, SearchSource } from 'ui/courier';
+import { IPrivate } from 'ui/private';
+import { IndexPatternEnhanced } from 'ui/index_patterns/_index_pattern';
+import { Filter } from '@kbn/es-query';
+import { reverseSortDir, SortDirection } from './utils/sorting';
+import { extractNanos, convertIsoToMillis } from './utils/date_conversion';
+import { fetchHitsInInterval } from './utils/fetch_hits_in_interval';
+import { generateIntervals } from './utils/generate_intervals';
+import { getEsQuerySearchAfter } from './utils/get_es_query_search_after';
+import { getEsQuerySort } from './utils/get_es_query_sort';
+
+export type SurrDocType = 'successors' | 'predecessors';
+export interface EsHitRecord {
+  fields: Record<string, any>;
+  sort: number[];
+  _source: Record<string, any>;
+}
+export type EsHitRecordList = EsHitRecord[];
+export interface IndexPatternGetProvider {
```

Comment:
Can be moved to `src/legacy/ui/public/index_patterns/_index_pattern.d.ts`

Reply:
Yes, thx!